app/controllers/memberships_controller.rb

```diff
@@ -0,0 +1,21 @@
+class MembershipsController < ApplicationController
+  def create
+    @membership = Membership.new(user: current_user, team_id: params[:team_id])
+
+    if @membership.save
+      redirect_to Team.find(params[:team_id]), notice: 'Team was successfully joined.'
+    else
+      render teams_path
```

Comment:
What if we redirected to the page where the user was at with a message stating that there was an error while trying to join the team? We can take advantage of `redirect_back` for this.